parsel/selector.py

```diff
@@ -44,11 +44,14 @@ def _st(st):
         raise ValueError('Invalid type: %s' % st)
 
 
-def create_root_node(text, parser_cls, base_url=None):
+def create_root_node(text, parser_cls, base_url=None, encoding='utf8'):
     """Create root node for text using given parser class.
     """
-    body = text.strip().replace('\x00', '').encode('utf8') or b'<html/>'
-    parser = parser_cls(recover=True, encoding='utf8')
+    if isinstance(text, bytes):
+        body = text
+    elif isinstance(text, six.text_type):
+        body = text.strip().replace('\x00', '').encode(encoding) or b'<html/>'
+    parser = parser_cls(recover=True, encoding=encoding)
```

Comment:
Why is `.strip().replace('\x00', '')` specific to strings?

Also, since in Python 2.6+ `bytes` is an alias for `str`, you are changing the behavior for Python 2. To simplify things, maybe we could remove Python 2 support from Parsel before moving on. Would you have time to work on a pull request for that?

Reply:
From my point of view - only with usage of `bytes` as input to Selector I can reduce number of extra memory allocations .

As result initially I focused to add support for `bytes` as input to `Selector` and at that moment I decided to... not touch current implementation for string processing.